src/icetransport.cpp

```diff
@@ -102,6 +106,12 @@ IceTransport::IceTransport(const Configuration &config, Description::Role role,
 					g_object_set(G_OBJECT(mNiceAgent.get()), "stun-server", nodebuffer, nullptr);
 					g_object_set(G_OBJECT(mNiceAgent.get()), "stun-server-port",
 					             std::stoul(servbuffer), nullptr);
+
+					// Do we need to set credentials for STUN server?
+					if (!server.username.empty() && !server.password.empty())
+						nice_agent_set_local_credentials(mNiceAgent.get(), mStreamId,
```

Comment:
I don't think this sets credentials for the STUN request. The function `nice_agent_set_local_credentials` actually forces the local ICE username fragment and password (the ones that are sent in the SDP and allow the remote peer to authenticate). STUN servers with credentials do not seem to be supported, probably because they are hardly ever deployed in practice.

Reply:
Actually I made tests before this commit.
But I realized now that also without credentials STUN server gives response. :)

I will delete 2. commit and will forget about STUN credentials for now.